util_fns.go

```diff
@@ -0,0 +1,289 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/olekukonko/tablewriter"
+)
+
+func getDevices() ([]string, error) {
+	var devices []string
+
+	devices_input := os.Getenv("devices_list")
+
+	if devices_input == "" {
```

Comment:
Do we need more validations here?
Do we want to try to type cast to a list of strings? etc..

Reply:
By default every input is a string in Bitrise, and any list is a line break. So the function which I'm using to get the text from input does this validation by default. Reference - [https://medium.com/golangspec/in-depth-introduction-to-bufio-scanner-in-golang-55483bb689b4](url)